expression-language.md

```diff
@@ -0,0 +1,369 @@
+# CloudEvents SQL Expression Language
+
+## Abstract
+
+The goal of this specification is to define an expression language SQL-like which can be used to express predicates on
+CloudEvents instances.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+
+- 1.1. [Conformance](#11-conformance)
+- 1.2. [Relation to Subscriptions API](#12-relation-to-the-subscriptions-api)
+
+2. [Language syntax](#2-language-syntax)
+
+- 2.1. [Expression](#21-expression)
+- 2.2. [Value identifiers and literals](#22-value-identifiers-and-literals)
+- 2.3. [Operators](#23-operators)
+- 2.4. [Functions invocation](#24-functions-invocation)
+
+3. [Language semantics](#3-language-semantics)
+
+- 3.1. [Type System](#31-type-system)
+- 3.2. [CloudEvent context identifiers and types](#32-cloudevent-context-identifiers-and-types)
+- 3.3. [Errors](#33-errors)
+- 3.4. [Operators](#34-operators)
+- 3.5. [Functions](#35-functions)
+- 3.6. [Evaluation of the expression](#36-evaluation-of-the-expression)
+
+4. [Examples](#4-examples)
+5. [References](#5-references)
+
+## 1. Introduction
+
+CloudEvents SQL expressions (also known as CESQL) allow matching of CloudEvent attributes against complex expressions
+that lean on the syntax of Structured Query Language (SQL) `WHERE` clauses. Using SQL-derived expressions for message
+filtering is in widespread implementation use because the Java Message Service (JMS) message selector syntax also leans
+on SQL. Note that neither the SQL standard (ISO 9075) nor the JMS standard nor any other SQL dialect are used as a
+normative foundation or constrain the expression syntax defined in this specification, but the syntax is informed by
```

Comment:
How far away from them are we? Is there a way to soften this statement?

Reply:
In support of Dug, I had thought that in discussion the plan for this path was to use an existing standard so as to be able to reuse existing implementations and knowledge.  Were there particular reasons for departing from a pre-existing canon?

Reply:
I think that's the major point of the discussion we had during the last meeting, where I pointed out the difference between the other JS like language and this language. We just can't claim we have a SQL subset, because otherwise:

* We have to choose **a specific SQL dialect** and **stick with it**: there are many out there and, in particular when it comes to built-ins, they all have different definitions.
* It gives us more freedom of movement when evolving this language: we might decide to add new features, maybe from another specific SQL dialect, that fits our needs. In particular when it comes down to _able to reuse existing implementations_, it literally means that we just can't move away from that particular standard and just use it as is (and evolve together with it).

Hence I believe this statement should not be soften, it should remain exactly as is to show our intention of giving you a SQL-like language, but not SQL-X or SQL-Y.